ipinfo/cmd_quota.go

```diff
@@ -0,0 +1,214 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"net/http"
+
+	"github.com/fatih/color"
+	"github.com/ipinfo/cli/lib/complete"
+	"github.com/ipinfo/cli/lib/complete/predict"
+	"github.com/spf13/pflag"
+)
+
+type QuotaRequests struct {
+	Day       int `json:"day"`
+	Month     int `json:"month"`
+	Limit     int `json:"limit"`
+	Remaining int `json:"remaining"`
+}
+
+type QuotaLimit struct {
+	Daily   int `json:"daily"`
+	Monthly int `json:"monthly"`
+}
+
+type QuotaFeatures struct {
+	Core          QuotaLimit `json:"core"`
+	Asn           QuotaLimit `json:"asn"`
+	BasicAsn      QuotaLimit `json:"basic_asn"`
+	Privacy       QuotaLimit `json:"privacy"`
+	Company       QuotaLimit `json:"company"`
+	Carrier       QuotaLimit `json:"carrier"`
+	Ranges        QuotaLimit `json:"ranges"`
+	Abuse         QuotaLimit `json:"abuse"`
+	HostedDomains QuotaLimit `json:"hosted_domains"`
+	Hostio        QuotaLimit `json:"hostio"`
+	Whois         QuotaLimit `json:"whois"`
+}
+
+type QuotaBody struct {
+	Token    string        `json:"token"`
+	Requests QuotaRequests `json:"requests"`
+	Features QuotaFeatures `json:"features"`
+}
+
+var completionsQuota = &complete.Command{
+	Flags: map[string]complete.Predictor{
+		"-d":         predict.Nothing,
+		"--detailed": predict.Nothing,
+		"-h":         predict.Nothing,
+		"--help":     predict.Nothing,
+	},
+}
+
+func printHelpQuota() {
+	fmt.Printf(
+		`Usage: %s quota [<opts>]
+
+Options:
+  --detailed, -d
+    show a detailed view of all available limits.
+    default: false.
+  --help, -h
+    show help.
+`, progBase)
+}
+
+func cmdQuota() error {
+	var fDetailed bool
+	pflag.BoolVarP(&fDetailed, "detailed", "d", false, "detail view.")
+	pflag.BoolVarP(&fHelp, "help", "h", false, "show help.")
+	pflag.Parse()
+
+	if fHelp {
+		printHelpQuota()
+		return nil
+	}
+
+	token := gConfig.Token
+	if token == "" {
+		return errors.New("no token available please login first")
+	}
+
+	res, err := http.Get("https://ipinfo.io/me?token=" + token)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+	quota := &QuotaBody{}
+	err = json.NewDecoder(res.Body).Decode(quota)
+	if err != nil {
+		return err
+	}
+	printStats(fDetailed, quota)
+
+	return nil
+}
+
+func getUsageBar(percentage int) string {
+	usageBar := "["
+	for i := 0; i < 100; i++ {
+		if i < percentage {
+			usageBar += "#"
+		} else {
+			usageBar += " "
+		}
+	}
+	usageBar += "]"
+	return fmt.Sprintf("%s %d%%\n", usageBar, percentage)
```

Comment:
Can we use `strings.Repeat` (https://pkg.go.dev/strings#Repeat) or similar helper funcs for the above logic?